command.php

```diff
@@ -178,16 +181,26 @@ private function checkCoreVulnerability()
             // Connect to wpvulndb
             $parameter = intval(str_replace('.', '', $coreVersion));
 
-            $client = new GuzzleHttp\Client();
-            $url = sprintf('https://wpvulndb.com/api/v2/wordpresses/%s', $parameter);
+            $cache = WP_CLI::get_cache();
+            $cache_key = "wp-sec/core.json";
+            $cache_file = $cache->has( $cache_key );
 
-            try {
-                $res = $client->request('GET', $url);
-            } catch (ClientException $e) {
-                WP_CLI::error(sprintf('Couldn\'t check wpvulndb @ %s', $url));
+            if ($cache_file && $this->cached) {
+                $json = json_decode($cache->read($cache_key), true);
             }
+            else {
+
+                $url = sprintf('https://wpvulndb.com/api/v2/wordpresses/%s', $parameter);
+                $req = WP_CLI\Utils\http_request('GET', $url);
+
+                if ( 20 != substr( $req->status_code, 0, 2 ) ) {
+                    WP_CLI::error(sprintf('Couldn\'t check wpvulndb @ %s (HTTP code %s)', $url, $req->status_code));
+                }
+
+                $cache->write($cache_key, $req->body);
```

Comment:
Well, I was thinking that too but then I realized that this could be the only way to _refresh/invalidate_ the cache (by a cron based job or whatever), other than adding another command line flag.
And indeed the overhead should be negligible.